src/Endpoint/IdentifierEndpoint.php

```diff
@@ -16,30 +17,22 @@
  *
  * @see EndpointBase
  * @see TokenAwareInterface
+ * @see ValidationAwareTrait
  */
 class IdentifierEndpoint extends EndpointBase implements TokenAwareInterface
 {
     use TokenAwareTrait;
     use ValidationAwareTrait;
 
-    protected string $correlationId;
-
-    public function getCorrelationId(): string
-    {
-        return $this->correlationId;
-    }
-
-    public function setCorrelationId(string $correlationId): self
+    public function getPermanentIdentifier(string $correlationId): string
     {
-        $this->correlationId = $correlationId;
-        return $this;
-    }
+        $url = $this->getEndpointUrl([':correlationId' => $correlationId]);
+        try {
+            $response = $this->rest->get($url, $this->getAuthorizationHeaders());
+        } catch (InvalidStatusCodeException $e) {
```

Comment:
I changed the exception catching. `Rest` always returns `InvalidStatusCodeException`, but some endpoints may change it into the validation exception.